File: android/app/src/main/java/com/anydaf/model/Bookmark.kt
package com.anydaf.model

import java.util.Date
import java.util.UUID

data class Bookmark(
    val id: UUID = UUID.randomUUID(),
    var name: String,
    var notes: String = "",
    val tractateIndex: Int,
    val daf: Int,
    val amud: Int,              // 0 = amud a, 1 = amud b
    val studySectionIndex: Int? = null,
    val createdAt: Date = Date()
) {
    val tractate: Tractate get() = allTractates[tractateIndex]
    val amudLabel: String get() = if (amud == 0) "a" else "b"
    val subtitle: String get() = "${tractate.name} Daf $daf$amudLabel"

    fun matches(query: String): Boolean {
        val q = query.lowercase()
        return name.lowercase().contains(q) || notes.lowercase().contains(q)
    }

    companion object {
        fun defaultName(tractateIndex: Int, daf: Int, amud: Int): String {
            val amudLabel = if (amud == 0) "a" else "b"
            return "${allTractates[tractateIndex].name} $daf$amudLabel"
        }
    }
}

File: AnyDaf/Bookmark.swift
import Foundation

struct Bookmark: Codable, Identifiable {
    let id: UUID
    var name: String
    var notes: String
    let tractateIndex: Int
    let daf: Int
    let amud: Int                   // 0 = amud a, 1 = amud b
    let studySectionIndex: Int?     // nil if bookmarked from main view
    let createdAt: Date

    var tractate: Tractate { allTractates[tractateIndex] }
    var amudLabel: String { amud == 0 ? "a" : "b" }
    var subtitle: String { "\(tractate.name) Daf \(daf)\(amudLabel)" }

    static func defaultName(tractateIndex: Int, daf: Int, amud: Int) -> String {
        "\(allTractates[tractateIndex].name) \(daf)\(amud == 0 ? "a" : "b")"
    }

    /// Returns true if this bookmark's name or notes contain the query (case-insensitive).
    func matches(_ query: String) -> Bool {
        let q = query.lowercased()
        return name.lowercased().contains(q) || notes.lowercased().contains(q)
    }
}
